dbms/src/Flash/Mpp/FineGrainedShuffleWriter.cpp

```diff
@@ -137,7 +137,7 @@ void FineGrainedShuffleWriter<ExchangeWriterPtr>::batchWriteFineGrainedShuffle()
         while (!blocks.empty())
         {
             const auto & block = blocks.back();
-            HashBaseWriterHelper::scatterColumnsInplace(block, num_bucket, collators, partition_key_containers_for_reuse, partition_col_ids, hash, selector, scattered);
+            HashBaseWriterHelper::scatterColumnsForFineGrainedShuffle(block, partition_col_ids, collators, partition_key_containers_for_reuse, partition_num, fine_grained_shuffle_stream_count, hash, selector, scattered);
```

Comment:
add assert to make sure that the real bucket num == fine_grained_shuffle_stream_count * partition_num?

Reply:
Since num_bucket is defined as "fine_grained_shuffle_stream_count * partition_num" in prepare() function, and CHECK "prepared" in write function already, I think it is safe here.

Reply:
ok.